features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/SessionReplayFeature.kt

```diff
@@ -193,30 +227,52 @@ internal class SessionReplayFeature(
                 InternalLogger.Target.USER,
                 { EVENT_MISSING_MANDATORY_FIELDS }
             )
-            return
+            return null
         }
 
-        if (currentRumSessionId.get() == sessionId) {
-            // we already handled this session
-            return
+        return SessionData(keepSession, sessionId)
+    }
+
+    @Suppress("ReturnCount")
+    private fun shouldHandleEvent(sessionData: SessionData): Boolean {
+        if (currentRumSessionId.get() != sessionData.sessionId) {
+            // sessionId changed so resample
+            isSessionSampledIn.set(rateBasedSampler.sample())
+        } else {
+            if (!recordingStateChanged.get()) {
+                // sessionId was already seen and recordingState did not change
+                return false
+            }
         }
 
         if (!checkIfInitialized()) {
-            return
+            return false
```

Comment:
shouldn't we move this above ?

Reply:
I'll move it, although it will require some changes to the unit tests as until now they didn't necessarily need to call onInitialize